### DISCUSSION TITLE
container create failed (`lstat /proc/63538/ns/ipc: no such file or directory`)

### DISCUSSION THREAD

**Discussion by clcc2019** [opencontainers/runc#4113] [Q&A]
### Description

container create failed: container_linux.go:338: creating new parent process caused "container_linux.go:1897: running lstat on namespace path \"/proc/63538/ns/ipc\" caused \"lstat /proc/63538/ns/ipc: no such file or directory\""

### Steps to reproduce the issue

1.
2.
3.

### Describe the results you received and expected

container create failed

### What version of runc are you using?

latest

### Host kernel information

4.19.90

**Comment by AkihiroSuda** [comment]
> lstat /proc/63538/ns/ipc: no such file or directory

What is your distro?
You need to recompile the kernel with `CONFIG_IPC_NS`.

**Comment by clcc2019** [comment]
> > lstat /proc/63538/ns/ipc: no such file or directory
> 
> What is your distro? You need to recompile the kernel with `CONFIG_IPC_NS`.

It supports CONFIG_IPC_NS because I use docker and runc 1.0-rc10 to run successfully

env: Kylin Linux Advanced Server V10 (Lance) ，4.19.90-52.23.v2207.ky10.x86_64
runc: v1.1.10
containerd: v1.7.8

**Comment by cyphar** [comment]
What does `ls -la /proc/self/ns` output?

**Reply by clcc2019** [reply]
<img width="684" alt="image" src="https://github.com/opencontainers/runc/assets/46099295/787901d9-bed5-4a31-a058-97cd69ed7ce4">

**Comment by cyphar** [comment]
It appears you're using a pre-1.1 version of runc (that error message was changed in e918d021399e62442711cc006f99beeae5e3ea1b, in 2021). Can you try to reproduce this issue with the latest version of runc (runc 1.1.9) or on the `main` branch?

Looking at whether the error comes from, it seems that you are creating a container using the namespace path of another process, and given the other information you have provided it seems that the process is dead so the namespace path doesn't exist when runc is trying to join it. It also looks like you're using Kubernetes, not runc directly (information that would've been helpful to include in the original issue description).

Can you provide some more information as to your setup and what you're trying to do when you get the error?
